types/src/traits/network.rs

```diff
@@ -109,105 +106,147 @@ pub enum NetworkError {
     ChannelSend,
     /// The underlying connection has been shut down
     ShutDown,
+    /// unable to cancel a request, the request has already been cancelled
+    UnableToCancel,
 }
 
-// TODO (da) make message anything that implements serializable
+/// id of message
+pub type RequestId = u64;
+
+/// the status of the message
+pub enum RequestStatus {
+    /// WIP
+    InProgress,
+    /// cancelled
+    Cancelled,
+    /// set successfully
+    Completed,
+    /// errored out
+    Error(NetworkError),
+    /// existance
+    DoesNotExist,
+}
 
-/// Describes, generically, the behaviors a networking implementation must have
+/// API for interacting directly with a consensus committee
+/// intended to be implemented for both DA and for validasting consensus committees
 #[async_trait]
-pub trait NetworkingImplementation<
+pub trait CommunicationChannel<
     TYPES: NodeType,
     LEAF: LeafType<NodeType = TYPES>,
     PROPOSAL: ProposalType<NodeType = TYPES>,
+    ELECTION: Election<TYPES>,
 >: Clone + Send + Sync + 'static
 {
     /// Returns true when node is successfully initialized
     /// into the network
-    ///
     /// Blocks until node is ready
-    async fn ready(&self) -> bool;
+    async fn ready_cc(&self) -> bool;
 
-    /// Broadcasts a message to the network
+    /// Shut down this network. Afterwards this network should no longer be used.
     ///
-    /// Should provide that the message eventually reach all non-faulty nodes
-    async fn broadcast_message(
+    /// This should also cause other functions to immediately return with a [`NetworkError`]
+    async fn shut_down_cc(&self) -> ();
+
+    /// broadcast message to those listening on the communication channel
+    /// non-blocking
+    async fn broadcast_message_cc(
         &self,
         message: Message<TYPES, LEAF, PROPOSAL>,
-    ) -> Result<(), NetworkError>;
+        election: &ELECTION,
+        view_number: TYPES::Time,
```

Comment:
Do we need the view number here? It can be retrieved from `message` anyway.

Reply:
Oooh yes 100%, I'll remove this 👍

Reply:
Actually it looks like like some cases don't have a view number? Specifically the `DataMessage::Transaction`. We'll probably need a view number for this if we just want to send it to the DA committee.

Reply:
If the view number is indeed needed when sending the transaction, maybe we can add `view_number` to the `DataMessage::SubmitTransaction` variant. So instead of `SubmitTransaction(TYPES::Transaction)`, we may have `SubmitTransaction(txn, view_number)`. I feel this is better than having a duplicate view number somewhere besides `Message`.

Reply:
I'll do that! I'm a bit concerned about adding slightly extra bandwidth in the future when we send the data messages over the network. 8 extra bytes per message is probably fine/something we can consider in the future.